weave/flow/obj.py

```diff
@@ -63,6 +64,23 @@ def from_uri(cls, uri: str, *, objectify: bool = True) -> Self:
             )
         return api.ref(uri).get(objectify=objectify)
 
+    def publish(self, name: Union[str, None] = None) -> ObjectRef:
+        # I'm using the `publish` term here for now, but ideally we call this `save`.
+        cls_name = self.__class__.__name__
+        if not is_registered(cls_name):
```

Comment:
this seems overly restrictive, why enforce this?

Reply:
you're right, we don't need it